aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/functional/CaseTypeTest.java

```diff
@@ -21,6 +26,17 @@ protected CaseTypeTest(AATHelper aat) {
     Supplier<RequestSpecification> asUserWithUser = asAutoTestCaseworkerWithUser();
     Supplier<RequestSpecification> asUser = asAutoTestCaseworker();
 
+    @BeforeAll
+    void setUp() {
+        Supplier<RequestSpecification> asTestImporter = asAutoTestImporter();
```

Comment:
Sateesh this is a good solution. I was wondering, as a further improvement, if we could avoid importing the same spreadsheet more than once. At the moment we do 1 here and 1 import in ImportDefinitionTest. Since basically all our func tests need a definition, can't we pull up both these import in the common superclass BaseTest?

Reply:
We can extract to a BaseTest superclass but each test method instance executes that. We need something like @BeforeAllTests - there is no built-in support atm. Will take a look into extensions..

Reply:
Implemented a custom annotation with BeforeAllCallback